make/compiler/Makefile.cray-prgenv

```diff
@@ -50,3 +50,22 @@ LIB_DYNAMIC_FLAG = -shared
 # will take care of that Since we want to be able to get other flags from e.g.
 # Makefile.intel, here we replace SPECIALIZE_CFLAGS with nothing.
 SPECIALIZE_CFLAGS =
+
+
+#
+# The CC and cc drivers use pkg-config to determine several things including
+# which libraries to link in. Set the appropriate libs based on our config:
+#
+
+PE_PKGCONFIG_PRODUCTS := PE_CHAPEL:$(PE_PKGCONFIG_PRODUCTS)
+PE_CHAPEL_MODULE_NAME := chapel
+
+# ugni and gasnet-{aries, gemini} require pmi and ugni
+ifeq ($(CHPL_MAKE_COMM), ugni)
+PE_CHAPEL_PKGCONFIG_LIBS := cray-pmi:cray-ugni:$(PE_CHAPEL_PKGCONFIG_LIBS)
+endif
+ifneq (,$(filter $(CHPL_MAKE_COMM_SUBSTRATE),gemini aries))
```

Comment:
Should we do this only ifeq ($(CHPL_MAKE_COMM), gasnet)?  I sometimes leave the subsidiary env vars set when I'm switching back and forth among different comm layers.  I'm not sure we bark if you have CHPL_COMM_SUBSTRATE=gemini with CHPL_COMM=none, but if we accept that combination then it would lead to the unnecessary inclusion of pmi and ugni.